src/Microsoft.AspNetCore.Razor.Evolution/DefaultRazorSourceLineCollection.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Microsoft.AspNetCore.Razor.Evolution
+{
+    internal class DefaultRazorSourceLineCollection : RazorSourceLineCollection
```

Comment:
A cut/paste to this new class in fact (because it is used in both `RazorSourceDocument` derived classes).
The only difference is in `GetLineStarts()`. The new code uses `RazorSourceDocument.Length` instead of `string.Length` and `RazorSourceDocument[index]` instead of `string[index]`.